apollo-router/src/plugins/telemetry/tracing/datadog_agent_span_processor.rs

```diff
@@ -0,0 +1,131 @@
+use opentelemetry_api::trace::SpanContext;
+use opentelemetry_api::trace::TraceResult;
+use opentelemetry_api::Context;
+use opentelemetry_sdk::export::trace::SpanData;
+use opentelemetry_sdk::trace::Span;
+use opentelemetry_sdk::trace::SpanProcessor;
+
+/// When using the Datadog agent we need spans to always be exported. However, the batch span processor will only export spans that are sampled.
+/// This wrapper will override the trace flags to always sample.
+/// THe datadog exporter itself will look at the `sampling.priority` trace context attribute to determine if the span should be sampled.
+#[derive(Debug)]
+pub(crate) struct DatadogBatchSpanProcessor<T: SpanProcessor> {
+    delegate: T,
+}
+
+impl<T: SpanProcessor> DatadogBatchSpanProcessor<T> {
+    pub(crate) fn new(delegate: T) -> Self {
+        Self { delegate }
+    }
+}
+
+impl<T: SpanProcessor> SpanProcessor for DatadogBatchSpanProcessor<T> {
+    fn on_start(&self, span: &mut Span, cx: &Context) {
+        self.delegate.on_start(span, cx)
+    }
+
+    fn on_end(&self, mut span: SpanData) {
+        let new_trace_flags = span.span_context.trace_flags().with_sampled(true);
```

Comment:
Should we check for either sampling priority to be set or measured to be set to set it to true ? I know for now we don't have that case but should we already try to handle this ?

Reply:
I had a long think about this. I think it's OK for the sole purpose is to change the sampled flag for exporting. I've refactored and renamed to reflect this: [a5efd62](https://github.com/apollographql/router/pull/6092/commits/a5efd6233589c7a2b8f23d25cd3661dc8ac2ac6a)